<launch>

  <arg name="image_raw_topic" default="/ackermann_vehicle/camera/rgb/image_raw" />
  <arg name="twist_cmd_topic" default="/cmd_vel" />
  <arg name="twist_linear_x" default="1" />
  <arg name="float_cmd_topic" default="" /> <!-- /flt_cmd -->
  <arg name="model" default=""/>
  <arg name="vel_cmd_topic" default="android_input_vel" />


  <group if="$(eval arg('model') != '')">
    <node name="driving" pkg="cnn" type="drive2a.py" output="screen" required="true" >
      <param name="image_raw_topic" value="$(arg image_raw_topic)" />
      <param name="twist_cmd_topic" value="$(arg twist_cmd_topic)" />
      <param name="twist_linear_x" value="$(arg twist_linear_x)" />
      <param name="float_cmd_topic" value="$(arg float_cmd_topic)" />
      <param name="modelname" value="$(arg model)" />
      <param name="vel_cmd_topic" value="$(arg vel_cmd_topic)" />
    </node>
  </group>
  
</launch>
